src/main/java/stacksandqueues/Stack.java

```diff
@@ -0,0 +1,38 @@
+package stacksandqueues;
+
+public class Stack {
+    public Node top;
+    public Node temp;
+
+    public Stack () {
+        this.top = null;
+    }
+
+    //if stack is null, make new value the top, else insert new value to new node.
+    public void push(int value) {
+        Node newNode = new Node(value);
+        if ( this.top == null) {
+            this.top = newNode;
+            newNode.next = null;
+        } else {
+            newNode.next = this.top;
+            this.top = newNode;
+        }
+    }
+
+    //remove node from top of stack and return node's value
+    public Node pop() {
+        if (top == null) {

```

Comment:
This function is returning the value of the top, it also needs to re-assign the top node it references.